diff --git a/app/Routes.tsx b/app/Routes.tsx
index 5dc29b3..5a5b2e7 100644
--- a/app/Routes.tsx
+++ b/app/Routes.tsx
@@ -5,7 +5,7 @@ import routes from './constants/routes.json';
 import App from './containers/App';
 import HomePage from './containers/HomePage';
 
-// Lazily load routes and code split with webpacck
+// Lazily load routes and code split with webpack
 const LazyCounterPage = React.lazy(() =>
   import(/* webpackChunkName: "CounterPage" */ './containers/CounterPage')
 );
@@ -16,11 +16,22 @@ const CounterPage = (props: Record<string, any>) => (
   </React.Suspense>
 );
 
+const LazyDistnetPage = React.lazy(() =>
+  import(/* webpackChunkName: "DistnetPage" */ './features/distnet/DistnetPage')
+);
+
+const DistnetPage = (props: Record<string, any>) => (
+  <React.Suspense fallback={<h1>Loading...</h1>}>
+    <LazyDistnetPage {...props} />
+  </React.Suspense>
+);
+
 export default function Routes() {
   return (
     <App>
       <Switch>
         <Route path={routes.COUNTER} component={CounterPage} />
+        <Route path={routes.DISTNET} component={DistnetPage} />
         <Route path={routes.HOME} component={HomePage} />
       </Switch>
     </App>
diff --git a/app/components/Home.tsx b/app/components/Home.tsx
index db007a9..5ce643e 100644
--- a/app/components/Home.tsx
+++ b/app/components/Home.tsx
@@ -1,13 +1,37 @@
+import _ from 'lodash';
 import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { Link } from 'react-router-dom';
 import routes from '../constants/routes.json';
 import styles from './Home.css';
+
+
 
 export default function Home(): JSX.Element {
+  const distnet = useSelector((state: RootState) => state.distnet);
+  const dispatch = useDispatch();
+
   return (
     <div className={styles.container} data-tid="container">
       <h2>Home</h2>
-      <Link to={routes.COUNTER}>to Counter</Link>
+      <Link to={routes.COUNTER}> -&gt; to Counter</Link>
+      <Link to={routes.DISTNET}> -&gt; to distnet settings</Link>
+
+      <div>
+        <ul>
+          {distnet.settings.sources &&
+            distnet.settings.sources.map((s) => <li key={s.id}>{s.name}</li>)}
+        </ul>
+
+        <ul>
+          {distnet.cache &&
+            _.map(distnet.cache, (value) => (
+              <li key={value}>{`${value.sourceId} -> ${value.localFile}`}</li>
+            ))}
+        </ul>
+
+
+      </div>
     </div>
   );
 }
diff --git a/app/constants/routes.json b/app/constants/routes.json
index 00770b0..e170b59 100644
--- a/app/constants/routes.json
+++ b/app/constants/routes.json
@@ -1,4 +1,5 @@
 {
   "HOME": "/",
-  "COUNTER": "/counter"
+  "COUNTER": "/counter",
+  "DISTNET": "/distnet"
 }
diff --git a/app/rootReducer.ts b/app/rootReducer.ts
index 35ea3e6..3a58bf1 100644
--- a/app/rootReducer.ts
+++ b/app/rootReducer.ts
@@ -3,10 +3,16 @@ import { connectRouter } from 'connected-react-router';
 import { History } from 'history';
 // eslint-disable-next-line import/no-cycle
 import counterReducer from './features/counter/counterSlice';
+// eslint-disable-next-line import/no-cycle
+import distnetReducer from './features/distnet/distnetSlice';
+// eslint-disable-next-line import/no-cycle
+
 
 export default function createRootReducer(history: History) {
   return combineReducers({
     router: connectRouter(history),
     counter: counterReducer,
+    distnet: distnetReducer,
+
   });
 }
diff --git a/package.json b/package.json
index ba8f633..05f7916 100644
--- a/package.json
+++ b/package.json
@@ -281,7 +281,10 @@
     "electron-debug": "^3.1.0",
     "electron-log": "^4.2.2",
     "electron-updater": "^4.3.1",
+    "env-paths": "^2.2.0",
     "history": "4.7.2",
+    "js-yaml": "^3.14.0",
+    "lodash": "^4.17.15",
     "react": "^16.13.1",
     "react-dom": "^16.12.0",
     "react-hot-loader": "^4.12.21",
